content/en/blog/_posts/2021-08-09-memory-manager-moves-to-beta.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: blog
+title: "Kubernetes Memory manager moves to beta"
+date: 2021-08-09
+slug: kubernetes-1-22-feature-memory-manager-moves-to-beta
+---
+
+**Authors:** Artyom Lukianov (Red Hat), Cezary Zukowski (Samsung)
+
+# Memory manager moves to beta
+
+The blog post describes the internals of the **Memory manager**, a beta feature of Kubernetes 1.22. The **Memory Manager** is a component in the **kubelet** ecosystem proposed to enable the feature of guaranteed memory (and hugepages) allocation for pods in the [Guaranteed QoS class](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#qos-classes).
+
+This blog post covers:
+
+1. Why do you need it?
+2. The internal details of how the **MemoryManager** works
+3. Current limitations of the **MemoryManager**
+4. Future work for the **MemoryManager**
+
+## Why do you need it?
+
+To get the best performance and latency for the workload container, container CPUs, PCI devices, and memory should be aligned to the same NUMA node. The kubelet already provides a set of managers to align CPUs and PCI devices, but you did not have a way to align memory. By default, the kernel will try to allocate the memory from the same NUMA node where container CPUs are placed, but it can not guarantee it.
```

Comment:
How about:
```suggestion
Some Kubernetes workloads run on nodes with
[non-uniform memory access](https://en.wikipedia.org/wiki/Non-uniform_memory_access) (NUMA).
If you have NUMA nodes in your cluster, you'll know about the potential for extra latency when
compute resources need to access memory that's on the same zone but a different NUMA
locality.

To get the best performance and latency for your workload, container CPUs,
peripheral devices, and memory should all be aligned to the same NUMA
locality.
Before Kubernetes v1.22, the kubelet already provided a set of managers to
align CPUs and PCI devices, but you did not have a way to align memory.
The Linux kernel was able to make best-effort attempts to allocate 
memory for tasks from the same NUMA node where the container is
executing are placed, but without any guarantee about that placement.
```
?

Reply:
Ah, my mistake. Actually “same zone” should say “same NUMA system” (I don't want to write _node_, as in the Kubernetes meaning, as that is liable to be confused with the Linux NUMA term “node”).
And “NUMA locality” could be changed to “NUMA cell” if you think that'd be better understood.

Reply:
I think we can use `machine` for the cluster node, WDT? I am just less familiar with the zone usage and for me, it looks like the data center terminology.